===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(rec_rset_group:deletion_map)) at 37 (line 800, column 24) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/src/rec-rset.c
+++ b/table1/benchmarks/recutils-1.8/src/rec-rset.c
@@ -801,6 +801,7 @@ rec_rset_group (rec_rset_t rset,
                                                record2,
                                                group_by))
                     {
+                      free(deletion_map);
                       /* Out of memory.  */
                       return NULL;
                     }
